File: scripts/browser/run-contextual.py
"""Run the frozen paired conversations through the native browser; retain all outcomes."""
import json,hashlib,pathlib,subprocess,sys
folder=pathlib.Path(__file__).parent
cases=(folder/'contextual-cases.json').read_text()
assert hashlib.sha256(cases.encode()).hexdigest()=='b69c46bfc4ddda87afbe57c3b225d2b59cecd613692b60dbd1884a736e93ae26'
arm=sys.argv[2]
assert arm in ['baseline','redesigned','fixture']
group=sys.argv[3] if len(sys.argv)>3 else 'core'
assert group in ['core','boundary','ui','smoke']
source=(folder/('contextual-'+group+'-checks.js')).read_text()
script=source.replace('__CASES__',cases).replace('__ARM__',json.dumps(arm))
wrapper=pathlib.Path.home()/'.codex/skills/playwright/scripts/playwright_cli.sh'
r=subprocess.run(['bash',str(wrapper),'-s='+sys.argv[1],'run-code',script],text=True,capture_output=True)
if '### Result\n' not in r.stdout:
 print(r.stdout);print(r.stderr);sys.exit(1)
report=json.loads(r.stdout.split('### Result\n',1)[1].split('\n### ',1)[0])
report['casesSha256']=hashlib.sha256(cases.encode()).hexdigest()
report['checkSourceSha256']=hashlib.sha256(source.encode()).hexdigest()
content=json.dumps(report,indent=2)+'\n'
out=pathlib.Path('artifacts/conversation')/('contextual-'+arm+'-'+hashlib.sha256(content.encode()).hexdigest()+'.json')
with out.open('x') as f:f.write(content)
out.chmod(0o444)
print(json.dumps({'artifact':str(out),'checks':len(report['results']),'failed':[r['name'] for r in report['results'] if not r['pass']],'error':report.get('error'),'responses':len(report.get('responses',[]))}))

File: scripts/browser/run-contextual.mjs
// Standalone fallback for native browser CLI batches that disconnect during
// long live runs. Execute the same frozen checks in a fresh, isolated browser.
import {readFile,writeFile,mkdir} from 'node:fs/promises';
import {createHash} from 'node:crypto';
import {pathToFileURL} from 'node:url';
const [runtime,group='core',revision]=process.argv.slice(2);
if(!runtime||!['core','boundary','copy_products'].includes(group)||!/[a-f0-9]{40}/.test(revision??''))throw new Error('Supply Playwright module, check group and full deployed revision.');
const sha=text=>createHash('sha256').update(text).digest('hex');
const cases=await readFile(new URL('contextual-cases.json',import.meta.url),'utf8');
if(sha(cases)!=='b69c46bfc4ddda87afbe57c3b225d2b59cecd613692b60dbd1884a736e93ae26')throw new Error('Frozen cases changed.');
const source=await readFile(new URL('contextual-'+group+'-checks.js',import.meta.url),'utf8');
const script=source.replace('__CASES__',cases).replace('__ARM__',JSON.stringify('redesigned'));
const {chromium}=await import(pathToFileURL(runtime).href),browser=await chromium.launch({channel:'chrome',headless:true});
try{
 await mkdir('output/playwright',{recursive:true});
 const page=await browser.newPage();await page.goto('https://case-resolution-agent.onrender.com/',{timeout:90000});
 const report=await(new Function('return ('+script+')'))()(page);
 Object.assign(report,{casesSha256:sha(cases),checkSourceSha256:sha(source),deployedRevision:revision,browserRunner:'Standalone isolated Playwright fallback; unchanged check source'});
 const body=JSON.stringify(report,null,2)+'\n',file='artifacts/conversation/contextual-redesigned-'+sha(body)+'.json';
 await writeFile(file,body,{flag:'wx',mode:0o444});
 console.log(JSON.stringify({artifact:file,checks:report.results.length,failed:report.results.filter(r=>!r.pass).map(r=>r.name),error:report.error,responses:report.responses?.length}));
 if(report.error||report.results.some(r=>!r.pass))process.exitCode=1;
}finally{await browser.close();}
